src/org/zaproxy/zap/extension/autoupdate/ManageAddOnsDialog.java

```diff
@@ -304,7 +320,56 @@ private JPanel getUninstalledAddOnsPanel() {
 		}
 		return uninstalledAddOnsPanel;
 	}
-	
+
+	private static JPanel createFilterPanel(final JXTable table) {
+		JPanel filterPanel = new JPanel();
+		filterPanel.setLayout(new GridBagLayout());
+
+		JLabel filterLabel = new JLabel(Constant.messages.getString("cfu.label.addons.filter"));
+		final JTextField filterTextField = new JTextField();
+
+		filterLabel.setLabelFor(filterTextField);
+		filterPanel.add(filterLabel, LayoutHelper.getGBC(0, 0, 1, 0.0D));
+		filterPanel.add(filterTextField, LayoutHelper.getGBC(1, 0, 1, 1.0D));
+
+		String tooltipText = Constant.messages.getString("cfu.label.addons.filter.tooltip");
+		filterLabel.setToolTipText(tooltipText);
+		filterTextField.setToolTipText(tooltipText);
+
+		// Set filter listener
+		filterTextField.getDocument().addDocumentListener(new DocumentListener() {
+			@Override
+			public void insertUpdate(DocumentEvent e) {
+				updateFilter();
+			}
+
+			@Override
+			public void removeUpdate(DocumentEvent e) {
+				updateFilter();
+			}
+
+			@Override
+			public void changedUpdate(DocumentEvent e) {
+				updateFilter();
+			}
+
+			public void updateFilter() {
+				String filterText = filterTextField.getText();
+				if (filterText.isEmpty()) {
+					table.setRowFilter(null);
+				} else {
+					try {
+						table.setRowFilter(RowFilter.regexFilter("(?i)" + filterText));
+					} catch (PatternSyntaxException e) {
+						String invalidPattern = Constant.messages.getString("cfu.label.addons.filter.invalidpattern");
+						filterTextField.setText(invalidPattern + filterText);

```

Comment:
This is leading to an exception:
```
65145 [AWT-EventQueue-0] ERROR org.zaproxy.zap.ZAP$UncaughtExceptionLogger  - Exception in thread "AWT-EventQueue-0"
java.lang.IllegalStateException: Attempt to mutate in notification
	at javax.swing.text.AbstractDocument.writeLock(AbstractDocument.java:1338)
	at javax.swing.text.AbstractDocument.replace(AbstractDocument.java:659)
	at javax.swing.text.JTextComponent.setText(JTextComponent.java:1718)
	at org.zaproxy.zap.extension.autoupdate.ManageAddOnsDialog$1.updateFilter(ManageAddOnsDialog.java:365)
	at org.zaproxy.zap.extension.autoupdate.ManageAddOnsDialog$1.insertUpdate(ManageAddOnsDialog.java:343)
```
I was thinking on a change of back/foreground colours, but an icon or even a message somewhere (under the text box?) is probably better.
Changing the text of the text box would also cause usability problems, since the user would have to delete the text set to fix the issues.